### PR TITLE
Debt token base library

### PR DESCRIPTION
**Pull request by msaug**
DebtTokenBase library, required for StableDebtToken and VariableDebtToken.
- Manages allowances for a debt token
- Associates the debt token to a specific underlying asset

### CODE REVIEW

File: contracts/protocol/tokenization/base/debt_token_base_library.cairo [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    func approve_delegation(delegatee : felt, amount : Uint256):
+        let (caller) = get_caller_address()
+        _approve_delegation(caller, delegatee, amount)
+    end
+
+    # @notice Returns the borrow allowance of the user
+    # @param from_user The user to giving allowance
+    # @param to_user The user to give allowance to
+    # @return The current allowance of `toUser`
+    func borrow_allowance(from_user : felt, to_user : felt):
```
Comment (hsouf) [inline contracts/protocol/tokenization/base/debt_token_base_library.cairo:54]: should return `allowance `

**Review by EdgarBarrantes — APPROVED** [review]
Looks good to me. I do think we should make a not, either in code (`delegation_with_sig` empty functions) or create an issue regarding the fact that this will (perhaps) make use of EIP712.

File: contracts/protocol/tokenization/base/debt_token_base_library.cairo [hunk 2]
```diff
@@ -0,0 +1,94 @@
+%lang starknet
+from starkware.cairo.common.cairo_builtins import HashBuiltin
```
Comment (achab) [inline contracts/protocol/tokenization/base/debt_token_base_library.cairo:2]: newline above

File: contracts/protocol/tokenization/base/debt_token_base_library.cairo [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    # @notice Delegates borrowing power to a user on the specific debt token.
+    # Delegation will still respect the liquidation constraints (even if delegated, a
+    # delegatee cannot force a delegator HF to go below 1)
+    # @param delegatee The address receiving the delegated borrowing power
+    # @param amount The maximum amount being delegated.
+    func approve_delegation{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        delegatee : felt, amount : Uint256
+    ):
+        let (caller) = get_caller_address()
+        return _approve_delegation(caller, delegatee, amount)
```
Comment (achab) [inline contracts/protocol/tokenization/base/debt_token_base_library.cairo:52]: why defining this internal function `_approve_delegation`?

**Review by achab — APPROVED** [review]
Clean PR, nothing to add. Ok to merge it. Thanks for that @msaug!

File: contracts/protocol/tokenization/base/debt_token_base_library.cairo (hunk 2 above)
Comment (EdgarBarrantes) [reply contracts/protocol/tokenization/base/debt_token_base_library.cairo:2]: It'd be cool to work on an import sorter extension for vscode, or just contribute to [the official](https://github.com/starkware-libs/cairo-lang/tree/master/src/starkware/cairo/lang/ide/vscode-cairo).

**Pull request merged**